### PR TITLE
adjust beacon timeout using delay value wp-media/wp-rocket--#6741

### LINKED ISSUE #6741: Adjust lcp-beacon.min.js timeout to be relative to `rocket_lcp_data.delay`
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version [x]
- Used the search feature to ensure that the bug hasn’t been reported before [x]

**Describe the bug**
If `rocket_lcp_delay` is superior to 10 seconds, that might be needed if a page has a heavy script that controls the layout, the beacon will return the right LCP but with a `failed` status. 

That failed status will prevent the buffer from adding OCI optimizations. 

This is caused by the `10` timeout limit located here:

https://github.com/wp-media/wp-rocket/blob/a8da7ae5fc1d1f524c4a9b625dc18456af176d7a/assets/js/lcp-beacon.js#L294-L296

Adjusting it to be relative to `rocket_lcp_data.delay` fixed the issue: 

```
window.rocket_lcp_data.delay + 10
```

**To Reproduce**
1. Add `rocket_lcp_delay ` with a value superior to 10
2. Check the beacon's AJAX request

**Expected behavior**
The timeout should take into consideration the LCP delay.

**Screenshots**
N/A

**Additional context**
Ticket - https://secure.helpscout.net/conversation/2615431939/495123?folderId=2952229

**Acceptance Criteria (for WP Media team use only)**
Clear instructions for developers, to be added before the grooming

### PR DESCRIPTION
**Pull request by Khadreal**
# Description
Adjust timeout, make the value dynamic

Fixes wp-media/wp-rocket#6741

### User documentation

Adjust the timeout to use the filter value from the backend.

### Technical documentation

*Explain how this code works. Diagram & drawings are welcomed.*

## Type of change
- [x] Enhancement (non-breaking change which improves an existing functionality).

## New dependencies
*List any new dependencies that are required for this change.*

## Risks
*List possible performance & security issues or risks, and explain how they have been mitigated.*

# Checklists

## Feature validation

- [ ] I validated all the Acceptance Criteria. If possible, provide sreenshots or videos.
- [ ] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [ ] I implemented built-in tests to cover the new/changed code.

## Documentation

- [ ] I prepared the user documentation for the feature/enhancement and shared it in the PR or the GitHub issue.
- [ ] The user documentation covers new/changed entry points (endpoints, WP hooks, configuration files, ...).
- [ ] I prepared the technical documentation if needed, and shared it in the PR or the GitHub issue.

## Code style
- [x] I wrote self-explanatory code about what it does.
- [ ] I wrote comments to explain why it does it.
- [x] I named variables and functions explicitely.
- [ ] I protected entry points against unexpected inputs.
- [x] I did not introduce unecessary complexity.
- [ ] I listed the introduced external dependencies explicitely on the PR.
- [ ] I validated the repo-specific guidelines from CONTRIBUTING.md.

### CODE REVIEW

**Review by Miraeld — DISMISSED** [review]

**Review by remyperona — DISMISSED** [review]

**Comment by MathieuLamiot** [comment]
@Khadreal There are conflicts to solve it seems, this branch should be updated against the trunk I guess.

Here are steps to be able to test the script with WP Rocket: https://wp-media.slack.com/archives/CUT7FLHF1/p1722526242449019?thread_ts=1722509646.276189&cid=CUT7FLHF1

File: dist/wpr-beacon.js [hunk 1]
```diff
@@ -288,7 +288,8 @@
         return this.errorCode;
       }
       const scriptTime = (/* @__PURE__ */ new Date() - this.scriptTimer) / 1e3;
-      if (10 <= scriptTime) {
+      const delay = this.config.delay + 10;
+      if (delay <= scriptTime) {
```
Comment (MathieuLamiot) [inline dist/wpr-beacon.js]: Change to remove.

**Review by MathieuLamiot — CHANGES_REQUESTED** [review]
It seems that the expected behavior is not clear based on the discussion you are having. The expected behavior is:
if the init method runs for more than 10 seconds, then the script status result is failed.

The sequence is as follows:
1. script starts and creates a new BeaconManager.
2. On DOMContentLoaded , the script waits for rocket_beacon_data.delay
3. After the delay, init() is called. 

The issue originally was that the timer was starting at point 1. Hence, the timer was running during the delay. This is something we don't want, hence the GH issue.

So, to solve this, the only thing we need is to do 
`this.scriptTimer = /* @__PURE__ */ new Date();`
at the beginning of init() instead of when instanciating `BeaconManager`

Therefore, we are sure the timer only runs during init() and the delay is not impacting it, regardless of its value.

**Comment by MathieuLamiot** [comment]
@DahmaniAdame to confirm my latest comment. But I think the original issue jumped directly into a solution (doing delay + 10) which caused the confusion, since there is a better way to handle this.

**Review by MathieuLamiot — APPROVED** [review]

**Comment by MathieuLamiot** [comment]
Changed base branch to develop. see https://wp-media.slack.com/archives/CUT7FLHF1/p1722593397186219?thread_ts=1722509646.276189&cid=CUT7FLHF1

**Review by hanna-meda — APPROVED** [review]
This PR looks good. Tested with steps from [TC](https://wpmediaqa.testrail.io/index.php?/cases/view/16423&group_by=cases:section_id&group_order=asc&display_deleted_cases=0&group_id=5875). 
Thank you.

**Pull request merged**